### PR TITLE
Task Annotation API 

### PR DESCRIPTION
**Pull request by geohacker**
This PR introduces the Task Annotation API:

* Introduces API end points to allow registered services to post JSON annotations
* And for the client to fetch annotations to render on the map
* Currently the map visualization [is hardcoded](https://github.com/hotosm/tasking-manager/pull/1289/files#diff-744e6c34c74ef9f175a41528b267f851R829) on `ml` type annotation and `building_area_diff` as the property, assuming the annotations are for building area prediction from an ML model. 
* The map allows to visualize annotations, and shows a legend.
* The API is [documented here](https://tasks-demo.hotosm.org/api-docs/swagger-ui/index.html?url=https://tasks-demo.hotosm.org/api/docs#!/project-admin/post_api_v1_project_project_id_task_annotations_annotation_type), and you can read the background [here](https://github.com/hotosm/tasking-manager/issues/1284) and [here](https://github.com/hotosm/tasking-manager/pull/1289).  
* Example of a [payload JSON is here](https://github.com/hotosm/tasking-manager/pull/1289#issuecomment-448905552).
* See a [demo task here](https://tasks-demo.hotosm.org/project/4662).

This PR also includes @ethan-nelson's application token work. 

cc @wronk @smit1678

### CODE REVIEW

**Comment by ethan-nelson** [comment]
sorry--wasn't clear that this is ready for review. i will do so this week!

**Comment by ethan-nelson** [comment]
Finishing up some testing still (I think there is a bug in the application key generation that doesn't allow deletion). We are also hopefully doing a develop->master merge soon, which will allow for a clean slate with some new changes. 

(If anyone else is testing this in the interim using the develop branch, you will need to change L14 of migrations/versions/05e1ecf9953a_.py to `3ee58dee57c9`.)

**Comment by ethan-nelson** [comment]
More updates. I used the online GitHub conflict resolution tool to update the `client/package.json` file--there was one package missing between tm4ml and develop, so that is now updated.

I've also isolated the issue with the inability to delete keys: the angular function was not defined properly. In addition, I found an error thrown on the sort parameter so I went ahead and removed that. Here is the diff for now--trying to figure out the easiest way to add a new commit.

```
diff --git a/client/app/profile/application-key.controller.js b/client/app/profile/application-key.controller.js
index 5f9f585..326e26d 100644
--- a/client/app/profile/application-key.controller.js
+++ b/client/app/profile/application-key.controller.js
@@ -69,7 +69,7 @@
          * Delete a user's application key
          * @param applicationKey
          */
-        function deleteApplicationKey(applicationKey){
+        vm.deleteApplicationKey = function(applicationKey){
             var resultsPromise = applicationKeyService.deleteApplicationKey(applicationKey);
             resultsPromise.then(function (data) {
                 vm.deleteResult = data
diff --git a/client/app/profile/application-key.html b/client/app/profile/application-key.html
index 50e3589..b04aec7 100644
--- a/client/app/profile/application-key.html
+++ b/client/app/profile/application-key.html
@@ -29,7 +29,7 @@
                                         <th><span>{{ 'Revoke key' | translate }}</span></th>
                                     </thead>
                                     <tbody>
-                                        <tr ng-repeat="item in appKeyCtrl.applications | orderBy: 'createdDate'">
+                                        <tr ng-repeat="item in appKeyCtrl.applications">
                                             <td>{{ item.createdDate }}</td>
                                             <td>{{ item.applicationkey }}</td>
                                             <td><a ng-click="appKeyCtrl.deleteApplicationKey(item.applicationkey)"><i>X</i></a></td>
```

**Comment by xamanu** [comment]
We have to reroll this on the latest `develop`.

**Comment by xamanu** [comment]
Rebased on latest `develop`, fixed conflicts on the way.

File: client/app/project/project.html [hunk 1]
```diff
@@ -1053,7 +1053,14 @@ <h6>{{ 'Options' | translate }}</h6>
                             </a>
                         </div>
                     </div>
-                    <div class="ol-control legend-control">
+                    <div ng-show="projectCtrl.projectData.annotations">
```
Comment (xamanu) [inline client/app/project/project.html]: Would it make sense to separate the general Task Annotation work and the one that addresses the specific application as an ML layer on the project page?

**Comment by xamanu** [comment]
CI fails because of deployment logic. I opened an issue #1488 to spinn up a new ml playground instance and deprecate the auto deployment on this branch. This shall fix the issue of the CI.

**Comment by ethan-nelson** [comment]
Thanks @xamanu

**Comment by xamanu** [comment]
Perhaps it makes sense to separate the backend work (#1577), that seems already to be at a good shape and could find a place into the main tasking manager code base. While the front-end needs much more work, still (#1578, #1579).

Would you be ok with closing this PR in favor of the follow ups, @geohacker?

**Comment by geohacker** [comment]
> Would you be ok with closing this PR in favor of the follow ups, @geohacker?

@xamanu sure thing. So it looks like to me that:
* backend code is now moved to #1577 - it'll be great to either merge this, or I can continue working on the improvements coming up over there.
* there's some ML visualisation as part of this PR, which I think is valuable to show in the short-term but it's your call whether to surface that or wait for #1578 and #1579.

**Comment by xamanu** [comment]
Great! Closing here.

> backend code is now moved to #1577 - it'll be great to either merge this, or I can continue working on the improvements coming up over there.

Yes, let's bring this into the code base.

> there's some ML visualisation as part of this PR, which I think is valuable to show in the short-term but it's your call whether to surface that or wait for #1578 and #1579.

Yes, we will do a PR with the front-end example. We can use this to deploy on the [tasks-assisted](https://tasks-assisted.hotosm.org) to have a running example. And then work on #1578 and #1579 for a general implementation that could become part of the main code base.

**Pull request closed without merging**